apps/iam/pkg/apis/iam/v0alpha1/register.go

```diff
@@ -0,0 +1,73 @@
+package v0alpha1
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	GROUP      = "iam.grafana.app"
+	VERSION    = "v0alpha1"
+	APIVERSION = GROUP + "/" + VERSION
+)
+
+var CoreRoleInfo = utils.NewResourceInfo(GROUP, VERSION,
+	"coreroles", "corerole", "CoreRole",
```

Comment:
IIUC, we have "Core roles" "Plugin roles" and "user defined roles" -- I think modeling each of these as their own resource will be very confusing.

Alternatively we can have a single "role" resource and an admission controller to say if people can edit it based on a property.

Reply:
Your point makes sense. There are currently two kinds `CoreRoles` (fixed and plugins), `Roles` (custom), well three if you count `GlobalRoles`.
I'm struggling a bit with the "how", though. "How do you wire the admission controller?", "How do you differentiate an entity that can edit the roles from one that cannot?".
The current approach has the benefit of making it extra clear to users that this is **not** something they can modify.

One other thing about CoreRole, its most likely that they will be a cluster scoped resource in the end. So if we end up merging, it will be with GlobalRole.